siquery_tables/src/windows/users.rs

```diff
@@ -0,0 +1,332 @@
+        lmcons::{MAX_PREFERRED_LENGTH, UNLEN, DNLEN},
+        winerror::*,
+    }
+};
+
+extern "system" {
+    fn LookupAccountSidW(
+        lpSystemName: LPCWSTR,
+        Sid: PSID,
+        Name: LPWSTR,
+        cchName: LPDWORD,
+        ReferencedDomainName: LPWSTR,
+        cchReferencedDomainName: LPDWORD,
+        peUse: PSID_NAME_USE
+    ) -> BOOL;
+}
+
+const NERR_SUCCESS: u32 = 0;
+const K_WELL_KNOWN_SIDS: [&'static str; 17 ] = [
+    "S-1-5-1",
+    "S-1-5-2",
+    "S-1-5-3",
+    "S-1-5-4",
+    "S-1-5-6",
+    "S-1-5-7",
+    "S-1-5-8",
+    "S-1-5-9",
+    "S-1-5-10",
+    "S-1-5-11",
+    "S-1-5-12",
+    "S-1-5-13",
+    "S-1-5-18",
+    "S-1-5-19",
+    "S-1-5-20",
+    "S-1-5-21",
+    "S-1-5-32",];
+
+impl Users {
+    pub(crate) fn new() -> Users {
+        Users {
+            uid: 0,
+            gid: 0,
+            uid_signed: 0,
+            gid_signed: 0,
+            username: String::new(),
+            description: String::new(),
+            directory: String::new(),
+            shell: String::new(),
+            uuid: String::new(),
+            type_: String::new(),
+        }
+    }
+
+    pub fn get_specific() -> Vec<Users> {
+        let mut users: Vec<Users> = Vec::new();
+        let mut processed_sid: Vec<String> = Vec::new();
+        process_local_accounts(&mut users, &mut processed_sid);
+        process_roaming_profiles(&mut users, &mut processed_sid);
+        users
+    }
+}
+
+fn process_local_accounts(users: &mut Vec<Users>, processed_sid: &mut Vec<String>) {
+    let dw_user_info_level: c_ulong = 3;
+
+    let mut dw_num_users_read_int = 0u32;
+    let dw_num_users_read: *mut c_ulong = &mut dw_num_users_read_int as *mut c_ulong;
+
+    let mut dw_total_users_int = 0u32;
+    let dw_total_users: *mut c_ulong = &mut dw_total_users_int as *mut c_ulong;
+
+    let mut resume_handle_int = 0u32;
+    let resume_handle: *mut c_ulong = &mut resume_handle_int as *mut c_ulong;
+
+    let mut user_buffer: Vec<u8> = Vec::with_capacity((MAX_PREFERRED_LENGTH) as usize);
+    loop {
+        let mut ret = unsafe {
+            NetUserEnum(ptr::null(),
+                        dw_user_info_level,
+                        0 as DWORD,
+                        user_buffer.as_mut_ptr() as *mut _ as *mut _,
+                        MAX_PREFERRED_LENGTH,
+                        dw_num_users_read,
+                        dw_total_users,
+                        resume_handle)
+        };
+
+        if (ret == NERR_SUCCESS || ret == ERROR_MORE_DATA) &&
+            user_buffer.as_mut_ptr() != ptr::null_mut() {
+
+            let mut iter_buff: LPUSER_INFO_3 = unsafe { ptr::read(user_buffer.as_mut_ptr() as *mut _) };
+
+            for _i in 0..unsafe { *dw_num_users_read }  {
+                let mut user = Users::new();
+                let mut dw_detailed_user_info_level: c_ulong = 4;
+                let mut user_lvl_4buff: Vec<*mut u8> = Vec::with_capacity((mem::size_of::<USER_INFO_4>()) as usize);

```

Comment:
Vec::<u8>::with_capacity((mem::size_of::<USER_INFO_4>()) as usize); 
is desired here to match the size of the USER_INFO_4 struct.
size_of() returns the number of bytes but Vec<*mut u8> is made of 64bit/32bit elements.